src/components/Homepage.jsx

```diff
@@ -11,25 +12,97 @@ import demoImage from "../images/demo-img.png";
 class Homepage extends Component {
   constructor(props) {
     super(props);
-    this.props = props;
+
+    const posts = this.loadPosts();
+    this.state = {
+      posts,
+      editingPostId: null,
+    };
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  loadPosts() {
+    const posts = [];
+    for (let i = 0; i < 4; i += 1) {
+      posts.push({
+        id: i,
+        username: "username",
+        postTime: new Date(),
+        imageSrc: demoImage,
+        content: "Some Content",
+      });
+    }
+
+    return posts;
+  }
+
+  handlePostCreation = (post) => {
+    // eslint-disable-next-line no-console
+    console.log(post);
+    // eslint-disable-next-line no-alert
+    alert("Post Created - Check Console");
+  }
+
+  handleEditToggle = (id) => {
```

Comment:
Consider we also need to be able to edit posts in Profile page, would it better to make a new component like postList and put all the functions related to edit posts there? That way we can handle editing in multiple pages.

Reply:
Yahh I sent this in the slack message to you but I'll put that as a separate PR but it shoudn't be hard to do.

Reply:
Oh right!! Sorry I totally forgot about that😖